packages/css/table.css

```diff
@@ -1,143 +1,149 @@
 
-.ds-table--lg {
-  --dsc-table-padding: var(--ds-spacing-3) var(--ds-spacing-3);
-}
-
-.ds-table__head {
-  z-index: 0;
-  box-sizing: border-box;
-  font-family: inherit;
+  border-collapse: separate; /* Using separate mode to enable border-radius */
+  border-radius: var(--dsc-table-border-radius);
   border-spacing: 0;
-  border-bottom: 2px solid var(--dsc-table-border-color);
-}
+  color: var(--dsc-table-color);
+  width: 100%;
 
-.ds-table__header__cell {
-  padding: var(--dsc-table-padding);
-  font-family: inherit;
-  background-color: var(--dsc-table-header-cell-background);
-  border-spacing: 0;
-  border-bottom: 2px solid var(--dsc-table-border-color);
-}
+  & > :is(tbody, thead) > tr > :is(th, td) {
+    background: var(--dsc-table-background);
+    border-bottom: var(--dsc-table-border);
+    padding: var(--dsc-table-padding);
+    text-align: inherit;
 
-.ds-table--sticky-header .ds-table__head .ds-table__header__cell {
-  position: sticky;
-  top: 0;
-  z-index: 1;
-}
+    &:is(th) {
+      font-weight: 500;
+    }
+  }
 
-.ds-table__header__cell--sortable {
-  padding: 0;
-}
+  & > thead > tr:last-child > :is(th, td) {
+    border-bottom: var(--dsc-table-header-divider);
+    background: var(--dsc-table-header-background);
+  }
 
-.ds-table__header__cell--sortable button {
-  position: relative;
-  width: 100%;
-  border: none;
-  font-family: inherit;
-  display: flex;
-  cursor: pointer;
-  gap: var(--ds-spacing-1);
-  align-items: center;
-  padding: var(--dsc-table-padding);
-  background-color: transparent;
-  color: var(--dsc-table-color);
-  z-index: 2;
-}
+  /* Add rounded border to first and last row */
+  & > :is(thead, tbody):first-child > tr:first-child > :is(th, td) {
+    &:first-child {
+      border-top-left-radius: var(--dsc-table-border-radius);
+    }
 
-.ds-table__header__cell--sorted button {
-  background-color: var(--dsc-table-header-sorted-background);
-}
+    &:last-child {
+      border-top-right-radius: var(--dsc-table-border-radius);
+    }
+  }
 
-.ds-table__header__cell--sortable button:focus-visible {
-  z-index: 3;
-  outline-offset: -3px;
-  box-shadow: unset;
-}
+  /* Add rounded border to last row */
+  & > :is(thead, tbody):last-child > tr:last-child > :is(th, td) {
+    &:first-child {
+      border-bottom-left-radius: var(--dsc-table-border-radius);
+    }
 
-.ds-table__header__cell--sortable button svg {
-  font-size: 1.2em;
-}
+    &:last-child {
+      border-bottom-right-radius: var(--dsc-table-border-radius);
+    }
+  }
 
-.ds-table__cell {
-  padding: var(--dsc-table-padding);
-  border-bottom: 1px solid var(--dsc-table-border-color);
-  background-color: var(--dsc-table-cell-backround);
-}
+  /**
+    * Sorting
+    */
+  & > thead > tr > [aria-sort] {
+    cursor: pointer;
+
+    & > button {
+      all: unset;
+      display: inline-block;
+      vertical-align: middle;
+
+      &:focus-visible {
+        box-shadow: var(--dsc-focus-boxShadow);
+      }
```

Comment:
Suggest changing this to our classname, and make the button fill so focus covers the whole `th`